WalletWasabi.Fluent/ViewModels/Wallets/Buy/Workflows/Workflow.cs

```diff
@@ -58,8 +60,17 @@ protected async Task ExecuteStepAsync(IWorkflowStep step)
 			catch (Exception ex)
 			{
 				step.Reset();
-				Logger.LogError($"An error occurred trying to execute Step '{step.GetType().Name}' in Workflow '{GetType().Name}'", ex);
-				OnStepError.SafeInvoke(this, ex);
+
+				// Only show error dialog for the first time, or if step requires user input
+				if (errorCount == 0 || step.IsInteractive)
+				{
+					OnStepError.SafeInvoke(this, ex);
+
+					// TODO: Roland, are we 100% sure we want to swallow errors and not log them?
```

Comment:
I think we should store the Exception instead of counting the error. So if the same exception keeps happening then only log it once. Meanwhile if there is a new exception we need to log that too.